digid_eherkenning/saml2/base.py

```diff
@@ -220,12 +221,17 @@ def create_config_dict(self, conf):
             "security": {
                 "signMetadata": True,
                 "authnRequestsSigned": True,
+                "logoutRequestSigned": True,
                 "wantAssertionsEncrypted": conf.get("want_assertions_encrypted", False),
                 "wantAssertionsSigned": conf.get("want_assertions_signed", False),
                 "soapClientKey": conf["key_file"],
                 "soapClientCert": conf["cert_file"],
                 "soapClientPassphrase": conf.get("key_passphrase", None),
-                "signatureAlgorithm": conf.get("signature_algorithm"),
+                # algorithm for requests with HTTP-redirect binding.
+                # AuthnRequest with HTTP-POST uses RSA_SHA256, which is hardcoded in OneLogin_Saml2_Auth.login_post
+                "signatureAlgorithm": conf.get(
+                    "signature_algorithm", OneLogin_Saml2_Constants.RSA_SHA1
```

Comment:
Is this to be consistent with the generate metadata value or is there another reason to add this default?

Reply:
There are two reasons:
1. consistence with metadata generation (as you said)
2. if `signature_algorithm` is not defined in `settings.DIGID` then [here](https://github.com/maykinmedia/python3-saml/blob/17c3655eeaa6e2018ac91b3c7b5c5d9b45986002/src/onelogin/saml2/auth.py#L618) in the `add_request_signature` method the input parameter `sign_algorithm` will be `None` and the code will crash. That's why we always need non-empty `security.signatureAlgorithm`

Reply:
Mhmm, I thought that python3-saml sets a default: https://github.com/maykinmedia/python3-saml/blob/maykin/src/onelogin/saml2/settings.py#L310 :thinking: But this default is not present when you call that function in the code line you pasted?

Reply:
I've commented this line and tried it and you are right, the default `RSA_SHA256` value is assigned. Perhaps before I did something wrong which crashed the code. 

Anyway as far as I understand SAML and Digid documentation we have to use `RSA_SHA1` or `DSA_SHA1` or HTTP-redirect binding. So I'd leave this default here

Reply:
Sounds good! Might be worth mentioning it in a comment in the Readme

Reply:
I've ended up documenting all Digid settings in the README :)